compliance-monitor/monitor.py

```diff
@@ -421,36 +430,42 @@ async def post_report(
     sep += len(SEP)
     signature = body_text[:sep - 1]  # do away with the ampersand!
     body_text = body_text[sep:]
-    json_text = None
+    try:
+        ssh_validate(keys, signature, body_text)
+    except ValueError:
+        raise HTTPException(status_code=401)
+
+    json_texts = []
     if content_type.endswith('-yaml'):
         yaml = ruamel.yaml.YAML(typ='safe')
-        document = yaml.load(body_text)
-        json_text = json.dumps(document, cls=TimestampEncoder)
+        documents = yaml.load_all(body_text)
+        json_texts = [json.dumps(document, cls=TimestampEncoder) for document in documents]
     elif content_type.endswith("-json"):
-        document = json.loads(body_text)
-        json_text = body_text
+        documents = [json.loads(body_text)]
+        json_texts = [body_text]
     else:
         raise HTTPException(status_code=500)
```

Comment:
maybe use `raise AssertionError("branch should never be reached")` instead, has same result but is clearer

Reply:
772b60390bfa214742e1b120fc5bf1a266bc2c41